pkg/deploy/lattice/target_group_manager.go

```diff
@@ -195,22 +197,28 @@ func (s *defaultTargetGroupManager) Delete(ctx context.Context, modelTg *model.T
 	}
 
 	if len(targetsToDeregister) > 0 {
-		deRegisterInput := vpclattice.DeregisterTargetsInput{
-			TargetGroupIdentifier: &modelTg.Status.Id,
-			Targets:               targetsToDeregister,
-		}
+		var deregisterTargetsError error
+		partitions := getPartitionedLatticeTargets(targetsToDeregister, 100)
+		for i, targets := range partitions {
+			deRegisterInput := vpclattice.DeregisterTargetsInput{
+				TargetGroupIdentifier: &modelTg.Status.Id,
+				Targets:               targets,
+			}
+			deRegResp, err := lattice.DeregisterTargetsWithContext(ctx, &deRegisterInput)
+			if err != nil {
+				return fmt.Errorf("Failed DeregisterTargets %s due to %s", modelTg.Status.Id, err)
+			}
+			isDeRegRespUnsuccessful := len(deRegResp.Unsuccessful) > 0
+			if isDeRegRespUnsuccessful {
+				deregisterTargetsError = errors.Join(deregisterTargetsError, fmt.Errorf("unsuccessful %d DeregisterTargets for partitions (%d/%d) for target group %s, deRegResp: %s",
+					len(deRegResp.Unsuccessful), i, len(partitions), modelTg.Status.Id, deRegResp))
 
-		deRegResp, err := lattice.DeregisterTargetsWithContext(ctx, &deRegisterInput)
-		if err != nil {
-			return fmt.Errorf("Failed DeregisterTargets %s due to %s", modelTg.Status.Id, err)
+			}
+			s.log.Infof("Success DeregisterTargets for partition (%d/%d) for target group %s", i+1, len(partitions), modelTg.Status.Id)
```

Comment:
This should be a debug log, don't need to see successful target deregistration on every partition in info mode

Reply:
Nit, clarity:
`Successfully deregistered targets to VPC Lattice Target Group %s for partition %d/%d`